File: business_logic/src/main/java/com/example/business_logic/service/RestaurantService.java
package com.example.business_logic.service;

import com.example.business_logic.model.Restaurant;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface RestaurantService {

    List<Restaurant> getAllRestaurants();

    Restaurant selectRestaurant(String name);
    int getMaxGuestNumber(String name);
    int getAvailableSpots(String name);

    void updateAvailableSpots(String name, int newAvailableSpots);
    Restaurant addRestaurant(int id, String name, String owner, String address, int availableSpots, int maximumGuestNumber);

}

File: cc_fe/cc_fe/src/app/services/restaurant.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {environment} from "../../environments/environment";
import {Observable} from "rxjs";
import {Restaurant} from "../model/restaurant";

@Injectable({
  providedIn: 'root'
})
export class RestaurantService {

  constructor(private httpClient: HttpClient) {
  }

  public getRestaurants(): Observable<any> {
    let url = environment.business_base_url + '/restaurants'
    return this.httpClient.get(url)
  }

  public addRestaurant(id: number, restaurant: Restaurant): Observable<any> {
    let url = environment.business_base_url + '/restaurants/add'
    let body = {
      "id": id,
      "name": restaurant.name,
      "owner": "Owner",
      "address": restaurant.address,
      "availableSpots": restaurant.availableSpots,
      // "maximumGuestNumber": restaurant.maximumGuestNumber,
      "imageUrl": restaurant.imageUrl
    }
    return this.httpClient.post(url, body)
  }
}
